code/parse/sexp/sexp_lookup.cpp

```diff
@@ -173,6 +173,14 @@ void dynamic_sexp_init()
 	auto& global = globals();
 	global.initialized = true;
 
+	// These sexps were added in a non determinate order (across different platforms and compilers). To ensure a
+	// consistent order for the same build across different platforms, we make the order consistent by sorting. This
+	// should make it possible to use dynamic SEXP in multi contexts.
```

Comment:
If the server and client are running different builds, and a sexp on one computer is missing on another computer, then the subsequent sexps will have different operator numbers.  This will prevent different builds from running any multiplayer missions even if neither one uses the actual sexp that is missing.

Reply:
Indeed, that is an issue I had not considered before.
What do you think about sending the string name of the operator instead of the number? Then it could be resolved back to the operator number on the client without running into inconsistency issues.
The next problem with that would be that this should only be done for the dynamic SEXPs though and it would still break the multi protocol with old builds but that shouldn't be too big of a problem.

Reply:
You could always make it part of the impending multi bump as well.

Reply:
Sending the sexp name would solve the problem, but I would rather not do that because it would roll back one of the major optimizations of the sexp system.  I can think of two good solutions, one simple and one more complex:

1) At mission load, the server sends the number of sexp operators.  If this number is different on any client, mission load fails.  (We can be virtually certain that two builds with the same *number* of operators will have the same *set* of operators.)

2) At mission load, create a hash map of operator numbers and send that to all clients.

I think I favor option #1.  Most of the time there will not be any difference in the number of operators, so a mapping would increase overhead (and it would also be a pain to code).  It would be a good compromise in that it allows players to use different builds while making sure they are still compatible.